db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/IndexRowGenerator.java

```diff
@@ -426,7 +426,7 @@ public boolean isUnique()
 
 	/** @see IndexDescriptor#baseColumnPositions */
 	public int[] baseColumnPositions() {
-		return id.baseColumnPositions();
+		return id == null ? null : id.baseColumnPositions();
```

Comment:
Wouldn't an index descriptor always have id set?  When can it happen that this is null?  It seems like an error condition.

Reply:
Right, this is super weird. Actually, no, `id` is not always set for `IndexRowGenerator`. I didn't debug into details about it, though.

Reply:
OK, well in any case, if we try to use a null baseColumnPositions later on it would end up with an NPE anyway, so adding this check probably doesn't hurt.